praktikumsplaner-frontend/frontend/src/components/ExcelImportNWK.vue

```diff
@@ -1,97 +1,142 @@
-                            :rules="rules"
-                            label="Fügen Sie eine Excel Datei ein"
+                <v-toolbar
+                    color="primary"
+                    dense
+                    dark
+                >
+                    <v-toolbar-title>Excel Import</v-toolbar-title></v-toolbar
+                >
+                <v-card-text>
+                    <span class="black--text">
+                        <v-icon
+                            x-large
+                            color="primary"
                         >
-                        </v-file-input>
-                    </v-list-item>
-                </v-list>
+                            {{ iconUpload }}
+                        </v-icon>
+                        {{ textUpload }}
+                    </span>
+                </v-card-text>
                 <v-card-actions>
                     <v-spacer />
                     <v-btn
                         text
+                        color="primary"
                         @click="cancel()"
                     >
                         Abbrechen
                     </v-btn>
                     <v-btn
                         color="primary"
-                        @click="uploadFile()"
-                    >
-                        Hochladen
+                        @click="handleFileImport()"
+                        >{{ buttonActionText }}
                     </v-btn>
+                    <input
+                        ref="uploader"
+                        class="d-none"
+                        type="file"
+                        :accept="excelFormat"
+                        @change="onFileChanged"
+                    />
                 </v-card-actions>
             </v-card>
         </v-form>
-    </v-dialog>
+        <v-btn
+            outlined
+            text
+            color="primary"
+            class="buttonEnd"
+            @click="cancel()"
+        >
+            Zurück
+        </v-btn>
+    </v-col>
 </template>
 
 <script setup lang="ts">
-import { ref } from "vue";
-import { useSnackbarStore } from "@/stores/snackbar";
-import { Levels } from "@/api/error";
+import { onMounted, ref } from "vue";
 import ExcelService from "@/api/ExcelService";
-import { useRules } from "@/composables/rules";
+import { EventBus } from "@/EventBus";
+import router from "@/router";
 
-const visible = ref<boolean>();
 const excelDatei = ref<File>();
-const form = ref<HTMLFormElement>();
-const snackbarStore = useSnackbarStore();
+const uploader = ref<HTMLInputElement>();
 const excelFormat =
     "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet";
-const validationRules = useRules();
-const rules = [
-    validationRules.fileTypeRule(
-        excelFormat,
-        "Es werden nur Excel Dateien akzeptiert"
-    ),
-];
+const buttonActionText = ref<string>("Auswählen");
+const isInUploadStage4 = ref<boolean>(false);
```

Comment:
Maybe find a better naming to express what this boolean is controlling (eg. isReadyToUpload or similar naming)